Mathlib/Algebra/SkewMonoidAlgebra/Basic.lean

```diff
@@ -160,6 +153,269 @@ theorem support_eq_empty {p} : p.support = ∅ ↔ (p : SkewMonoidAlgebra k G) =
 
 end Support
 
+section Coeff
+
+/-- `coeff f a` (often denoted `f.coeff a`) is the coefficient of `a` in `f`. -/
+def coeff : SkewMonoidAlgebra k G → G → k
+  | ⟨p⟩ => p
+
+@[simp]
+theorem coeff_ofFinsupp (p) : coeff (⟨p⟩ : SkewMonoidAlgebra k G) = p := rfl
+
+theorem coeff_injective : Injective (coeff : SkewMonoidAlgebra k G → G → k) := by
+  rintro ⟨p⟩ ⟨q⟩
+  simp only [coeff, DFunLike.coe_fn_eq, imp_self, ofFinsupp.injEq]
+
+@[simp]
+theorem coeff_inj (p q : SkewMonoidAlgebra k G) : p.coeff = q.coeff ↔ p = q :=
+  coeff_injective.eq_iff
+
+theorem toFinsupp_apply (f : SkewMonoidAlgebra k G) (g) : f.toFinsupp g = f.coeff g := rfl
+
+@[simp]
+theorem coeff_zero (g : G) : coeff (0 : SkewMonoidAlgebra k G) g = 0 := rfl
+
+@[simp]
+theorem mem_support_iff {f : SkewMonoidAlgebra k G} {a : G} : a ∈ f.support ↔ f.coeff a ≠ 0 := by
+  rcases f with ⟨⟩
+  simp only [coeff, support_ofFinsupp, Finsupp.mem_support_iff, ne_eq, implies_true]
+
+theorem not_mem_support_iff {f : SkewMonoidAlgebra k G} {a : G} :
+    a ∉ f.support ↔ f.coeff a = 0 := by
+  simp only [mem_support_iff, ne_eq, not_not]
+
+theorem ext_iff {p q : SkewMonoidAlgebra k G} : p = q ↔ ∀ n, coeff p n = coeff q n := by
+  rcases p with ⟨f : G →₀ k⟩
+  rcases q with ⟨g : G →₀ k⟩
+  simpa [coeff] using DFunLike.ext_iff (f := f) (g := g)
+
+@[ext]
+theorem ext {p q : SkewMonoidAlgebra k G} : (∀ a, coeff p a = coeff q a) → p = q := ext_iff.2
+
+@[simp]
+theorem coeff_add (p q : SkewMonoidAlgebra k G) (a : G) :
+    coeff (p + q) a = coeff p a + coeff q a := by
+  rcases p
+  rcases q
+  simp_rw [← ofFinsupp_add, coeff]
+  exact Finsupp.add_apply _ _ _
+
+@[simp]
+theorem coeff_smul {S} [SMulZeroClass S k] (r : S) (p : SkewMonoidAlgebra k G) (a : G) :
+    coeff (r • p) a = r • coeff p a := by
+  rcases p
+  simp_rw [← ofFinsupp_smul, coeff]
+  exact Finsupp.smul_apply _ _ _
+
+end Coeff
+
+section Single
+
+/-- `single a b` is the finitely supported function with value `b` at `a` and zero otherwise. -/
+def single (a : G) (b : k) : SkewMonoidAlgebra k G := ⟨Finsupp.single a b⟩
+
+@[simp]
+theorem toFinsupp_single (a : G) (b : k) : (single a b).toFinsupp = Finsupp.single a b := rfl
+
+@[simp]
+theorem ofFinsupp_single (a : G) (b : k) : ⟨Finsupp.single a b⟩ = single a b := rfl
+
+theorem coeff_single {a a' : G} {b : k} [Decidable (a = a')] :
+    coeff (single a b) a' = if a = a' then b else 0 := by
```

Comment:
I think I'd slightly prefer the statement
```suggestion
theorem coeff_single {a : G} {b : k} [DecidableEq G] :
    coeff (single a b) = Pi.single a b := by
```
since there are lots of lemmas about `Pi.single`.

Reply:
The lemma follows [`Finsupp.single_apply`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Finsupp/Defs.html#Finsupp.single_apply) and [`MonoidAlgebra.single_apply`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/MonoidAlgebra/Defs.html#MonoidAlgebra.single_apply). `Finsupp` also has [`Finsupp.single_eq_pi_single`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Finsupp/Defs.html#Finsupp.single_eq_pi_single), maybe it should just be added instead of replacing `coeff_single`?